EIPS/eip-5748.md

```diff
@@ -0,0 +1,191 @@
+---
+eip: 5748
+title: Approval Expiration for EIP-20 Tokens
+description: This standard introduces a standardized way for users to set expiration limits on ERC-20 token approvals.
+author: Go+ Security (@GoPlusSecurity), Jeff Hall (@Mr-Lucky), Xavi (@XaaaaavitheFool), Turan Vural (@turanzv)
+discussions-to: <https://ethereum-magicians.org/t/eip-5748-automatic-expiration-of-token-approvals/>
+status: Draft
+type: Standards Track
+category: Interface
+created: 2022-10-02) format
+requires:
+---
+
+This standard introduces a standard way for users to set expiration limits on ERC-20 token approvals to prevent the loss of assets due to contractrual vulnerabilities or authorization attacks.
+
+## Abstract
+This standard adds an expiration time to the `_approve()` function of token contracts to automatically recall approvals within a certain time period. This functionality as a standard will prevent vulnerabilty exploits due to outstanding token approvals.
```

Comment:
This proposal doesn't add an expiration time, but rather a **validity period** after the transaction is included on-chain. I know it's a pedantic difference, but it's important from a security point of view.